packages/client/src/runtime/core/types/Extensions.ts

```diff
@@ -120,12 +120,15 @@ export type DynamicModelExtensionThis<TypeMap extends TypeMapDef, M extends Prop
   [P in keyof ExtArgs['model'][Uncapitalize<M & string>]]:
     Return<ExtArgs['model'][Uncapitalize<M & string>][P]>
 } & {
-  [P in Exclude<keyof TypeMap['model'][M], keyof ExtArgs['model'][Uncapitalize<M & string>]>]:
-    {} extends TypeMap['model'][M][P]['args'] // will match fully optional args
+  [P in Exclude<keyof TypeMap['model'][M]['operations'], keyof ExtArgs['model'][Uncapitalize<M & string>]>]:
+    {} extends TypeMap['model'][M]['operations'][P]['args'] // will match fully optional args
     ? <A extends TypeMap['model'][M][P]['args']>(args?: A) =>
-        PrismaPromise<GetOperationResult<TypeMap['model'][M][P]['payload'], A, P & Operation>>
-    : <A extends TypeMap['model'][M][P]['args']>(args: A) =>
-        PrismaPromise<GetOperationResult<TypeMap['model'][M][P]['payload'], A, P & Operation>>
+        PrismaPromise<GetOperationResult<TypeMap['model'][M]['operations'][P]['payload'], A, P & Operation>>
+    : <A extends TypeMap['model'][M]['operations'][P]['args']>(args: A) =>
+        PrismaPromise<GetOperationResult<TypeMap['model'][M]['operations'][P]['payload'], A, P & Operation>>
+} & {
+  [P in Exclude<'fields' & keyof TypeMap['model'][M], keyof ExtArgs['model'][Uncapitalize<M & string>]>]:
+    TypeMap['model'][M]['fields'] // TODO remove & keyof TypeMap['model'][M] once fieldRefs is GA
```

Comment:
```suggestion
    TypeMap['model'][M]['fields'] // TODO remove & keyof TypeMap['model'][M] once fieldReference is GA
```
So we can find this later easily via a search